### PR TITLE
fix: [DHIS2-13190][DHIS2-13196] tei geometry

### PR DESCRIPTION
**Pull request by simonadomnisoru**

### CODE REVIEW

**Review by superskip — CHANGES_REQUESTED** [review]
Hi! Looks like most of the volume in this PR is due to adding line breaks in comma separated list, which I guess is fine. Got a couple of comments. The largest comment asks you to add a lookup table somewhere, I have left it to you to decide the specifics of where and how.

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/RenderFoundation.js [hunk 1]
```diff
@@ -3,18 +3,22 @@
...
 import { buildDataElement, buildTetFeatureType } from './DataElement';
 import { getProgramTrackedEntityAttributes, getTrackedEntityTypeId } from '../helpers';
 
-const getFeatureType = (featureType: ?string) => (featureType ? capitalizeFirstLetter(featureType.toLowerCase()) : 'None');
+const getFeatureType = (featureType: ?string) =>
+    (featureType ? capitalizeFirstLetter(featureType.toLowerCase()) : 'None');
 
-const buildTetFeatureTypeField = (trackedEntityType: CachedTrackedEntityType) => {
+const buildProgramSection = programSection =>
+    programSection.trackedEntityAttributes.reduce((acc, { id }) => [...acc, id], []);
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/RenderFoundation.js]: Good catch! I changed it to `map`.

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/hooks/useGeometryValues.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    useEffect(() => {
+        if (geometry && ['POLYGON', 'POINT'].includes(featureType)) {
+            const dataElementType =
+                featureType === dataElementTypes.POLYGON ? dataElementTypes.POLYGON : dataElementTypes.COORDINATE;
+            const clientValue = convertServerToClient(geometry.coordinates, dataElementType);
+            const formValue = convertClientToForm(clientValue, dataElementType);
+
+            setFormValues({ [`FEATURETYPE_${featureType}`]: formValue });
+            setClientValues({ [`FEATURETYPE_${featureType}`]: clientValue });
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetProfile/DataEntry/hooks/useGeometryValues.js]: Sure, I implemented a lookup table for the geometry features logic. Thank you for the suggestion! 😄

**Review by superskip — COMMENTED** [review]
Looks good! :+1:

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/helpers/geometry.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const DATAELEMENTTYPE = {
+    POINT: dataElementTypes.COORDINATE,
+    POLYGON: dataElementTypes.POLYGON,
+};
+
+const LABELS = {
+    POINT: 'Coordinate',
+    POLYGON: 'Area',
+};
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetProfile/DataEntry/helpers/geometry.js]: Hey, 
In the end, it seemed easier and cleaner to merge all the values into a single const object. I also added the flow types. 
Thanks!

**Review by superskip — COMMENTED** [review]
Hey! I found more to comment on.. This is the third cycle on the same PR, I hope that does not mean I will never be satisfied :stuck_out_tongue:

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/DataElement.js [hunk 4]
```diff
@@ -188,13 +189,13 @@ const buildOptionSet = async (
 
 export const buildTetFeatureType = (featureType: 'POINT' | 'POLYGON') => {
     const dataElement = new DataElement((dataEntry) => {
-        dataEntry.id = `FEATURETYPE_${featureType}`;
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/DataElement.js:191]: Hey,

As I explained in https://github.com/dhis2/capture-app/pull/2808#discussion_r914718990 the `WidgetProfile` is self-contained, that is why the definition of `GEOMETRY` have to be inside the folder. 
IMHO optimizations for other parts of the app are out of scope for this ticket. 
Thanks!

File: src/core_modules/capture-core/components/WidgetProfile/DataEntry/helpers/geometry.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// @flow
+import { dataElementTypes } from '../../../../metaData';
+
+export const GEOMETRY: {
+    POINT: { FEATURETYPE: string, DATAELEMENTTYPE: string, LABEL: string },
+    POLYGON: { FEATURETYPE: string, DATAELEMENTTYPE: string, LABEL: string },
+} = {
+    POINT: {
+        FEATURETYPE: 'FEATURETYPE_POINT',
```
[the remark this answers is not in the document]
Comment (superskip) [reply src/core_modules/capture-core/components/WidgetProfile/DataEntry/helpers/geometry.js:8]: It makes sense then, I suppose. The widgets will all depend on the tracker API though, and therefore _could_ import from some common module defining this dependency. I believe no such module has been made yet however, and that is definitely not part of this ticket to introduce.

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.39 and 2.38.2 versions

**Pull request merged**